tools/testing/test_selections.py

```diff
@@ -30,43 +32,75 @@
         NUM_PROCS = 1
 
 
+class ShardedTest(NamedTuple):
+    name: str
+    shard: int
+    num_shards: int
+    time: Optional[float]
```

Comment:
As I see you always default `test.time` to 0 when it is not defined.  Why not just set this default to 0 instead of marking as `Optional`.  With this, all the `test.time or 0` statement could just become test.time.  Another way would be to add something like `get_time()` in `ShardedTest` and do the None check there

Reply:
To me, 0 is different from None since 0 means the test actually took 0 seconds (or some really really small amount of time), while None means we don't have information about how long the test took
added get_time func